plugin/src/main/java/org/jenkinsci/plugins/structs/describable/DescribableModel.java

```diff
@@ -427,6 +422,29 @@ private Object coerce(String context, Type type, Object o) throws Exception {
         }
     }
 
+    private Object coerceStringToNumber(@Nonnull Class numberClass, @Nonnull String o) throws ClassCastException {
+        try {
+            if (numberClass.equals(Integer.class)) {
+                return Integer.valueOf(o);
+            } else if (numberClass.equals(Float.class)) {
+                return Float.valueOf(o);
+            } else if (numberClass.equals(Double.class)) {
+                return Double.valueOf(o);
+            } else if (numberClass.equals(Long.class)) {
+                return Long.valueOf(o);
+            } else if (numberClass.equals(Byte.class)) {
+                return Byte.valueOf(o);
+            } else if (numberClass.equals(Short.class)) {
+                return Short.valueOf(o);
+            } else {
+                // Fallback for any other Number - just return the original string.
+                return o;
+            }
+        } catch (NumberFormatException nfe) {
+            throw new ClassCastException(nfe.getMessage());
```

Comment:
nit: I think it would be nice to include the `context` from the caller in the message here, and to not box `numberClass` in the argument to this method so we could have a very clear exception message along the lines of:

    throw new ClassCastException(context + " expects " + numberClass + " but was unable to coerce the received value \"" + o + "\" to that type");

Reply:
Works for me.